### PR TITLE
[improvement]: Graph memory support for non-structured models.

### PR DESCRIPTION
**Pull request by spike-spiegel-21**
Issue: #1822 

The current graph memory tools have the following:
https://github.com/mem0ai/mem0/blob/965f7a373554433f9f83b666f65abaa7d817e611/mem0/graphs/tools.py#L120

This is good for structured-models of openai but gives error for others.

This is an attempt to change the tool for non-structured models. 

### Refrences
For Azure openai: [here](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/structured-outputs?tabs=python-secure)
_(Support for structured outputs was first added in API version [2024-08-01-preview](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-08-01-preview/inference.json).)_

For openai: [here](https://platform.openai.com/docs/guides/structured-outputs/introduction)
_However, Structured Outputs with response_format: {type: "json_schema", ...} is only supported with the gpt-4o-mini, gpt-4o-mini-2024-07-18, and gpt-4o-2024-08-06 model snapshots and later._

### CODE REVIEW

**Comment by spike-spiegel-21** [comment]
Hi, @prateekchhikara I will be adding tests in some time. Please check the initial commits.

File: mem0/memory/graph_memory.py [hunk 1]
```diff
@@ -67,9 +71,13 @@ def add(self, data, filters):
 
         update_memory_prompt = get_update_memory_messages(search_output, extracted_entities)
 
+        _tools=[UPDATE_MEMORY_TOOL_GRAPH, ADD_MEMORY_TOOL_GRAPH, NOOP_TOOL]
+        if self.llm_provider == ["azure_openai_structured","openai_structured"]:
```
Comment (prateekchhikara) [inline mem0/memory/graph_memory.py:75]: this should be `if self.llm_provider in ["azure_openai_structured","openai_structured"]:`

Comment (prateekchhikara) [reply mem0/memory/graph_memory.py:75]: @spike-spiegel-21 can you please check this?

Comment (spike-spiegel-21) [reply mem0/memory/graph_memory.py]: done @prateekchhikara

**Comment by prateekchhikara** [comment]
@spike-spiegel-21 can you resolve the conflicts, they are minor

**Comment by spike-spiegel-21** [comment]
> @spike-spiegel-21 can you resolve the conflicts, they are minor

Merge conflicts resolved @prateekchhikara

**Pull request merged**